go-controller/pkg/ovn/base_network_controller_policy.go

```diff
@@ -1111,12 +1112,25 @@ func (bnc *BaseNetworkController) addNetworkPolicy(policy *knet.NetworkPolicy) e
 
 	// 1. save required namespace information to use for netpol create,
 	npKey := getPolicyKey(policy)
-	nsInfo, nsUnlock := bnc.getNamespaceLocked(policy.Namespace, true)
-	if nsInfo == nil {
-		return fmt.Errorf("unable to get namespace for network policy %s: namespace doesn't exist", npKey)
+
+	var aclLogging ACLLoggingLevels
+	if bnc.doesNetworkRequireIPAM() {
+		nsInfo, nsUnlock := bnc.getNamespaceLocked(policy.Namespace, true)
+		if nsInfo == nil {
+			return fmt.Errorf("unable to get namespace for network policy %s: namespace doesn't exist", npKey)
+		}
+		aclLogging = nsInfo.aclLogging
+		nsUnlock()
+	} else {
+		// we haven't kept track of the namespace in the cache; we need to load
+		// it from the informer. The nsInfo is cached as a side-effect.
+		nsInfo, nsUnlock, err := bnc.loadNamespace(policy.Namespace)
+		if err != nil {
+			return fmt.Errorf("failed to get nsInfo for namespace %s", policy.Namespace)
+		}
+		aclLogging = nsInfo.aclLogging
+		nsUnlock()
```

Comment:
I am bit confused here. 

I guess that if you just used `getNamespaceLocked` it wouldn't work because you would get back `nil` from it as there is no cached namespace info. But we are still watching for namespace events. So I guess that when we process those events, an error is happening preventing the cached namespace info from being populated, and is being kept retried to no avail. All for nothing because we don't expect that ever to succeed? Which means that we are watching namespaces for nothing and that we are not reacting namespace updates.

So I  think we should just use `getNamespaceLocked` and fix whatever issue is preventing the cache from being populated as a result of processing namespace events. I think the issue is just taking care of not checking for pod IPs on `ensureNamespaceLockedForSecondaryNetwork` and wherever `addPodToNamespaceForSecondaryNetwork` is being called from (the latter is already being handled).

Reply:
> I am bit confused here.
> 
> I guess that if you just used `getNamespaceLocked` it wouldn't work because you would get back `nil` from it as there is no cached namespace info. But we are still watching for namespace events. So I guess that when we process those events, an error is happening preventing the cached namespace info from being populated, and is being kept retried to no avail. All for nothing because we don't expect that ever to succeed? Which means that we are watching namespaces for nothing and that we are not reacting namespace updates.
> 
> So I think we should just use `getNamespaceLocked` and fix whatever issue is preventing the cache from being populated as a result of processing namespace events. I think the issue is just taking care of not checking for pod IPs on `ensureNamespaceLockedForSecondaryNetwork` and wherever `addPodToNamespaceForSecondaryNetwork` is being called from (the latter is already being handled).

It's not only the IPs; I really don't see a need for even creating the address set (since afaiu, it's only purpose is to hold IPs that won't be there).

I would prefer to create another function similar to `ensureNamespaceLockedForSecondaryNetwork` that doesn't care about IPs, and doesn't create the address set. 
@jcaamano

Reply:
Trying a different approach that I think meets our ideas half-way. 

I agree this is more readable, and simpler to reason about.
@jcaamano